core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -505,10 +529,24 @@ public static <E extends Entry<Key,Value>> TabletMetadata convertRow(Iterator<E>
               BulkFileColumnFamily.getBulkLoadTid(val));
           break;
         case CurrentLocationColumnFamily.STR_NAME:
-          te.setLocationOnce(val, qual, LocationType.CURRENT);
+          try {
+            te.setLocationOnce(val, qual, LocationType.CURRENT);
+          } catch (IllegalStateException e) {
+            te.futureAndCurrentLocationSet = true;
+            if (!suppressLocationError) {
+              throw e;
+            }
+          }
```

Comment:
May be able to push checking `suppressLocationError` and setting `suppressLocationError` into `setLocationOnce` and can avoid catching the exception here. Something like the following.

```java
  private void setLocationOnce(String val, String qual, LocationType lt, boolean suppressError) {
    if (location != null) {
      if(suppressError) {
        futureAndCurrentLocationSet = true;
      } else {
        throw new IllegalStateException("Attempted to set second location for tableId: " + tableId
                + " endrow: " + endRow + " -- " + location + " " + qual + " " + val);
      }
    }
    location = new Location(val, qual, lt);
  }
```

Reply:
Moved the error suppression logic in 9153726